### PR TITLE
[4.2] Change task plugin to service provider

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
Changes the requests task plugin to service provider. Additionally some untranslated strings are fixed and the manifest file is cleaned up.

The change in the `TaskPluginTrait.php` is needed, so forms are loaded from the forms folder in the root of the plugin.

### Testing Instructions
- Create a GET request task in the task scheduler extension
- Run the task manually

### Actual result BEFORE applying this Pull Request
All works.

### Expected result AFTER applying this Pull Request
All works.

### CODE REVIEW

File: plugins/task/requests/src/Extension/Requests.php [hunk 1]
```diff
@@ -65,6 +61,44 @@ public static function getSubscribedEvents(): array
...
+	 *
+	 * @var    CMSApplicationInterface
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected $app;
+
+	/**
+	 * The http factory
+	 *
+	 * @var    HttpHttpFactory
```
Comment (brianteeman) [inline plugins/task/requests/src/Extension/Requests.php]: is this a typo ?

File: plugins/task/requests/src/Extension/Requests.php [hunk 2]
```diff
@@ -65,6 +61,44 @@ public static function getSubscribedEvents(): array
...
+	 * @since  __DEPLOY_VERSION__
+	 */
+	private $httpFactory;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param   DispatcherInterface  $dispatcher   The dispatcher
+	 * @param   array                $config       An optional associative array of configuration settings
+	 * @param   HttpHttpFactory      $httpFactory  The user factory
```
Comment (brianteeman) [inline plugins/task/requests/src/Extension/Requests.php]: same typo question

File: plugins/task/requests/src/Extension/Requests.php (hunk 1 above)
Comment (laoneo) [reply plugins/task/requests/src/Extension/Requests.php]: yes it is

**Comment by richard67** [comment]
~~Somehow this PR seems to show unrelated changes on GitHub, as if the branch had been rebased.~~

Sorry, was looking at the wrong PR.

File: administrator/components/com_scheduler/src/Traits/TaskPluginTrait.php [hunk 3]
```diff
@@ -149,6 +148,13 @@ public function enhanceTaskItemForm($context, $data = null): bool
 		$path                = \dirname((new \ReflectionClass(static::class))->getFileName());
 		$enhancementFormFile = $path . '/forms/' . $enhancementFormName . '.xml';
 
+		// Service provider based plugins need a different path lookup
+		if (!file_exists($enhancementFormFile))
+		{
+            $path                = JPATH_PLUGINS . '/' . $this->_type . '/' . $this->_name;
```
Comment (richard67) [inline administrator/components/com_scheduler/src/Traits/TaskPluginTrait.php]: Please check code style (indentation) here.

Comment (laoneo) [reply administrator/components/com_scheduler/src/Traits/TaskPluginTrait.php]: I'm already on PSR-12 :smirk:

Comment (richard67) [reply administrator/components/com_scheduler/src/Traits/TaskPluginTrait.php]: Even psr-12 does not allow that mess. Please look again at the code and adjust your editor so you can see what is a space and what is a tabulator.

File: plugins/task/requests/src/Extension/Requests.php [hunk 4]
```diff
@@ -65,6 +61,44 @@ public static function getSubscribedEvents(): array
...
+	 * @since  __DEPLOY_VERSION__
+	 */
+	private $httpFactory;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param   DispatcherInterface  $dispatcher   The dispatcher
+	 * @param   array                $config       An optional associative array of configuration settings
+	 * @param   HttpFactory      $httpFactory  The user factory
```
Comment (heelc29) [inline plugins/task/requests/src/Extension/Requests.php]:
```suggestion
	 * @param   DispatcherInterface  $dispatcher   The dispatcher
	 * @param   array                $config       An optional associative array of configuration settings
	 * @param   HttpFactory          $httpFactory  The http factory
```

**Comment by heelc29** [comment]
Can you remove the version attribute?
https://github.com/joomla/joomla-cms/blob/4d793c9fa73f47a6d0680323035fcda6f60b7a84/plugins/task/requests/requests.xml#L2

**Comment by laoneo** [comment]
done

File: administrator/components/com_scheduler/src/Traits/TaskPluginTrait.php [hunk 5]
```diff
@@ -149,6 +148,13 @@ public function enhanceTaskItemForm($context, $data = null): bool
 		$path                = \dirname((new \ReflectionClass(static::class))->getFileName());
 		$enhancementFormFile = $path . '/forms/' . $enhancementFormName . '.xml';
 
+		// Service provider based plugins need a different path lookup
```
Comment (HLeithner) [inline administrator/components/com_scheduler/src/Traits/TaskPluginTrait.php]: Is this path different path to the path used before?
I mean in the end it's
`/plugins/system/name/forms/form.xml`
or is it now on a different position?

Comment (laoneo) [reply administrator/components/com_scheduler/src/Traits/TaskPluginTrait.php]: It's the same path at the end, this change fixes it for namespaced classes. When the extension class is namespaced, then the old way points to the src/Extension directory.

Comment (laoneo) [reply administrator/components/com_scheduler/src/Traits/TaskPluginTrait.php]: Actually the old way should be removed...

Comment (HLeithner) [reply administrator/components/com_scheduler/src/Traits/TaskPluginTrait.php]: that's what I was getting at.

Comment (laoneo) [reply administrator/components/com_scheduler/src/Traits/TaskPluginTrait.php]: Removed the reflection code

**Comment by roland-d** [comment]
Thanks everybody

**Pull request merged**